sdk/search/azure-search/src/main/java/com/azure/search/DocumentResponseConversions.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.exception.ResourceNotFoundException;
+
+/**
+ * Internal utility class for Document Response conversions.
+ */
+class DocumentResponseConversions {
+
+    private static final String ODATA_CONTEXT = "@odata.context";
+
+    /**
+     * Map exceptions to be more informative
+     *
+     * @param throwable to convert
+     * @return Throwable
+     */
+    static Throwable exceptionMapper(Throwable throwable) {
+        if (throwable instanceof HttpResponseException
+            && throwable.getMessage().equals("Status code 404, (empty body)")) {
+            return new ResourceNotFoundException("Document not found",
+                ((HttpResponseException) throwable).getResponse());
+        }
+        return throwable;
+    }
+
+    /**
+     * Drop fields that shouldn't be in the returned object
+     *
+     * @param document document object
+     */
+    static void cleanupDocument(Document document) {
```

Comment:
It looks like the Java SDK is always sending `odata.metadata=minimal` instead of `odata.metadata=none`, causing extra annotations like this to appear in the response. `odata.metadata=minimal` is only necessary for APIs that return polymorphic models (e.g. -- scoring functions, data detection policies, skills, etc.). The .NET SDK sets the `Accept` header to `application/json;odata.metadata=none` for all "document" requests (Autocomplete, Index, Lookup, Search, Suggest, Count). We need to change the Java SDK to do the same in a subsequent PR so we can get rid of hacks like these and save space in the response payload.